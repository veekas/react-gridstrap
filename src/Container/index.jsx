import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Container = ({ children, className, fluid, ...props }) => {
  const ContainerDiv = styled('div')`
    box-sizing: border-box;
    position: relative;
    display: grid;
    width: ${fluid ? '100%' : 'auto'};
    grid-template-columns: repeat(12, 1fr);
    grid-gap: 0;
  `;

  return (
    <ContainerDiv className={className} {...props}>
      {children}
    </ContainerDiv>
  );
}

Container.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  fluid: PropTypes.bool,
};

Container.defaultProps = {
  children: null,
  className: null,
  fluid: false,
};

export default Container;
