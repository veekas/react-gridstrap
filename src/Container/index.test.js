import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Container from './';

describe('Container', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Container />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<Container />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
