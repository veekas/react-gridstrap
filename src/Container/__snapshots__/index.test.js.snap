// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container matches snapshot 1`] = `
<div
  className="css-1dg94ii"
/>
`;
